proposals/handler_test.go

```diff
@@ -783,8 +783,8 @@ func TestProposal_DuplicateTXs(t *testing.T) {
 			require.Equal(t, p.Ballot.ID(), ballot.ID())
 			return true, nil
 		})
-	th.mm.EXPECT().AddBallot(&p.Ballot).Return(nil).DoAndReturn(
-		func(got *types.Ballot) error {
+	th.mm.EXPECT().AddBallot(context.TODO(), &p.Ballot).Return(nil).DoAndReturn(
```

Comment:
What's the meaning of `.Return(nil).DoAndReturn(...)`? Does it return `nil` the first time or is `Return(nil)` simply overwritten? If the latter, then:
```suggestion
	th.mm.EXPECT().AddBallot(context.TODO(), &p.Ballot).DoAndReturn(
```

Reply:
not intentional. by the fact that tests were passing, i'd say it was overwritten. fixed.